### PR TITLE
Elastic Agent ACI compliant image

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

Bringing back work which allows 
- image to run on Azure Container Instances (no root group)
- run cloud defend (heartbeat capabilities set)
- smaller size (got rid of double chown and reduced size of npm modules setup for synthethics)
- allow container to run as different user (get rid of world permission stripping)

Size comparison
```
branch
docker images | grep cloud
docker.elastic.co/beats-ci/elastic-agent-cloud. 8.12.0-SNAPSHOT  331c23d10995  5 minutes ago. 3.08GB

main
docker images | grep cloud
docker.elastic.co/beats-ci/elastic-agent-cloud  8.12.0-SNAPSHOT  a6e4e5b9ff81  6 minutes ago. 3.08GB
```

- Closes https://github.com/elastic/elastic-agent/issues/3711
- Closes https://github.com/elastic/elastic-agent/issues/82

### CODE REVIEW

**Comment by michalpristas** [comment]
still working on -complete image size

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl [hunk 1]
```diff
@@ -6,14 +6,25 @@
...
+{{- end}}
+    # ECE needs to create config here under non-1000 user
+    chmod 0777 {{ $beatHome}} && \
+    mkdir -p {{ $beatHome }}/data {{ $beatHome }}/data/elastic-agent-{{ commit_short }}/logs && \
     find {{ $beatHome }} -type d -exec chmod 0755 {} \; && \
     find {{ $beatHome }} -type f -exec chmod 0644 {} \; && \
-    find {{ $beatHome }}/data -type d -exec chmod 0770 {} \; && \
-    find {{ $beatHome }}/data -type f -exec chmod 0660 {} \; && \
+    find {{ $beatHome }}/data -type d -exec chmod 0777 {} \; && \
+    find {{ $beatHome }}/data -type f -exec chmod 0666 {} \; && \
```
Comment (pchila) [inline dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:27]: Why do we need to create the user and the correct ownership in the intermediate stage ? Wouldn't it be enough to copy with the correct ownership and user in the last stage (line ~127) ?

**Review by pchila — APPROVED** [review]
Aside from some unnecessary change of ownership in the earlier stage, the change looks good.

Just out of curiosity: how hard would it be to always create the `elastic-agent` user and group, associate all the files to that and as last instruction add a USER instruction ? (ideally with the value being an argument instead of a template)

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl (hunk 1 above)
Comment (michalpristas) [reply dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:27]: hmm that;s a nice idea, will check

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl [hunk 2]
```diff
@@ -27,7 +38,7 @@ RUN mkdir -p {{ $beatHome }}/data {{ $beatHome }}/data/elastic-agent-{{ commit_s
     (chmod 0755 {{ $beatHome }}/data/elastic-agent-*/components/pf-elastic-collector || true) && \
     (chmod 0755 {{ $beatHome }}/data/elastic-agent-*/components/pf-elastic-symbolizer || true) && \
     (chmod 0755 {{ $beatHome }}/data/elastic-agent-*/components/pf-host-agent || true) && \
-    find {{ $beatHome }}/data/elastic-agent-{{ commit_short }}/components -name "*.yml*" -type f -exec chown root:root {} \; && \
+    find {{ $beatHome }}/data/elastic-agent-{{ commit_short }}/components -name "*.yml*" -type f -exec chown {{ .user }}:{{ .user }} {} \; && \
```
Comment (michalpristas) [inline dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:41]: this is a no, modules permissions are checked to be owned by root

**Comment by cmacknz** [comment]
Did we confirm this image can be deployed to Elastic Cloud successfully and that ECE stack can come up.

I would also like @blakerouse to review this before merge to see if there is anything that will cause problems for the non-root/unprivileged agent work.

**Comment by michalpristas** [comment]
i created cloud instance using our regular testing/cloud env. this was fine. 
for the use case lastly reported i haven't found a way to deploy custom image (QA labs does not seem to work)

**Comment by michalpristas** [comment]
i'm seeing some weird behavior but i see it on main as well. 
i need to test this on another machine to rule out my env. i don't trust my mac

**Comment by michalpristas** [comment]
seems like some env issues on my side, rebuilt on my linux machine and it seems to produce system metrics normally.

**Review by blakerouse — APPROVED** [review]
Looks good to me

**Pull request merged**